azure-iot-device/tests/common/test_http_transport.py

```diff
@@ -0,0 +1,279 @@
+
+        transport.request(fake_method, fake_path, mocker.MagicMock()).result()
+        assert mock_http_client_constructor.call_count == 2
+
+    @pytest.mark.it("Uses the HTTP Transport SSL Context.")
+    def test_uses_ssl_context(self, mocker, transport, mock_http_client_constructor):
+        ssl_context_mocker = transport._ssl_context
+        done = transport.request(fake_method, fake_path, mocker.MagicMock())
+        done.result()
+
+        assert mock_http_client_constructor.call_count == 1
+        assert mock_http_client_constructor.call_args[1]["context"] == ssl_context_mocker
+
+    @pytest.mark.it("Formats the request URL correctly given a path and no query parameters.")
+    def test_formats_http_client_request_with_only_method_and_path(
+        self, mocker, transport, mock_http_client_constructor
+    ):
+        mock_http_client_request = mock_http_client_constructor.return_value.request
+        fake_method = "__fake_method__"
+        fake_path = "__fake_path__"
+        expected_url = "https://{}/{}".format(fake_hostname, fake_path)
+        done = transport.request(fake_method, fake_path, mocker.MagicMock())
+        done.result()
+
+        assert mock_http_client_constructor.call_count == 1
+        assert mock_http_client_request.call_count == 1
+        assert mock_http_client_request.call_args[0][0] == fake_method
+        assert mock_http_client_request.call_args[0][1] == expected_url
+        assert mock_http_client_request.call_args[1]["body"] == ""
+        assert bool(mock_http_client_request.call_args[1]["headers"]) is False
+
+    @pytest.mark.it("Formats the request URL correctly given a path and query parameters.")
+    def test_formats_http_client_request_with_method_path_and_query_params(
+        self, mocker, transport, mock_http_client_constructor
+    ):
+        mock_http_client_request = mock_http_client_constructor.return_value.request
+        fake_method = "__fake_method__"
+        fake_path = "__fake_path__"
+        fake_query_params = "__fake_query_params__"
+        expected_url = "https://{}/{}?{}".format(fake_hostname, fake_path, fake_query_params)
+        # output = {"response": None, "error": None}
+
+        # def request_callback(response=None, error=None):
+
+        done = transport.request(
+            fake_method, fake_path, mocker.MagicMock(), query_params=fake_query_params
+        )
+        done.result()
+        assert mock_http_client_constructor.call_count == 1
+        assert mock_http_client_request.call_count == 1
+        assert mock_http_client_request.call_args[0][0] == fake_method
+        assert mock_http_client_request.call_args[0][1] == expected_url
+        assert mock_http_client_request.call_args[1]["body"] == ""
+        assert bool(mock_http_client_request.call_args[1]["headers"]) is False
+
+    @pytest.mark.it(
+        "Calls HTTP Client Connection request method with correct method, url, body, and headers."
```

Comment:
>Connection [](start = 27, length = 10)

"Connection" makes this more confusing than it needs to be.

Also, I try to avoid referring to specific methods, and instead write a requirement based on what it does, to avoid coupling requirements with implementation (thus avoiding some requirement rot) - Something like "Sends an HTTP Request via the HTTP Client" is probably better. Listing the args, and the method name is unnecessary to understanding what this requirement is.